key.go

```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package core
+package otel
```

Comment:
We discussed a little at the Go SIG meeting that maybe `KeyValue` could go to it own package. It seems, by looking at the gitter chat, that people prefer a `kv` package, changing the UX to `kv.String("key", "value")` or  `"kv.Int("num", 1)`.

Reply:
+1 that `kv.String(...)` is more understandable.